Helper/AlgoliaHelper.php

```diff
@@ -12,6 +12,9 @@
 
 class AlgoliaHelper extends AbstractHelper
 {
+    /** @var array */
+    protected $nonCastableAttributes = ['sku', 'name', 'description'];
```

Comment:
Can you make it `private` and definite setter / getter method(s)?

Reply:
Not really. Getter/setter are anti-patterns.

Reply:
Using accessors only adds bloatware in this case. Not to mention that accessors aren't widely used in Magento and `private`'s are considered evil by a lot of Magento devs. We write a lot of Magento extensions and generally make everything public since there is no good reason to make some protected or private. 

Keeping it protected allows the developer who wants to do a change to the extension upon the same variable instead of having to pluginize accessors to change one small line.

Reply:
Here's my go-to-link on the subject; http://www.yegor256.com/2014/09/16/getters-and-setters-are-evil.html